test/distributed/fsdp/test_fsdp_param_exec_order_wrap.py

```diff
@@ -58,26 +64,66 @@ def device(self):
         [ShardingStrategy.FULL_SHARD, ShardingStrategy.SHARD_GRAD_OP],
     )
     @parametrize("iters", [1, 3])
-    def test_fsdp_flatten_params_exec_order(self, sharding_strategy: ShardingStrategy, iters: int):
+    def test_fsdp_flatten_params_exec_order(
+        self,
+        sharding_strategy: ShardingStrategy,
+        iters: int,
+    ):
         """Tests the basic APIs of FSDP with ParamExecOrderWrapPolicy"""
-        fsdp_model = Model.wrap(sharding_strategy, self.device)
+        wrap_policy = ParamExecOrderWrapPolicy(init_policy=always_wrap_policy)
+        fsdp_model = Model.wrap(sharding_strategy, self.device, wrap_policy=wrap_policy)
         self.assertTrue(fsdp_model._is_param_exec_order_prep_stage())
         for _ in range(iters):
             input = fsdp_model.module.get_input(self.device)
             output = fsdp_model(*input)
             loss = fsdp_model.module.get_loss(input, output).to(self.device)
             loss.backward()
         params_list = list(fsdp_model.parameters())
+        # Since the forward execution order is NOT consistent with
+        # the model definition order, the ordering in flatten_named_params_exec_order
+        # should be different from named_parameters.
+        self.assertEqual(
+            fsdp_model._fsdp_params_exec_order,
+            [params_list[0], params_list[2], params_list[3], params_list[1]],
+        )
+        self.assertTrue(fsdp_model._use_param_exec_order_policy())
+        self.assertTrue(not fsdp_model._is_param_exec_order_prep_stage())
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize(
+        "sharding_strategy",
+        [ShardingStrategy.FULL_SHARD, ShardingStrategy.SHARD_GRAD_OP],
+    )
+    @parametrize("iters", [0, 1])
+    def test_fsdp_flatten_params_exec_order_symbolic_trace(
```

Comment:
To intercept, @LinjianMa  do not worry about this for now 😅 
I can take care of it in one sweeping refactor. I am still figuring out the best way to introduce utilities for making the subtesting as frictionless as possible. (I am currently working on this for `test_fsdp_core.py` and `test_fsdp_mixed_precision.py` since they take the longest, and I will have a PR out soon.)